Mathlib/Topology/Algebra/InfiniteSum/NatInt.lean

```diff
@@ -335,3 +272,222 @@ theorem Summable.tendsto_atTop_zero {f : ℕ → G} (hf : Summable f) : Tendsto
+  -- Note this is much easier to prove if you assume more about the target space, but we have to
+  -- work hard to prove it under the very minimal assumptions here.
+  apply (hf.add (hasSum_ite_eq (0 : ℤ) (f 0))).hasSum_of_sum_eq fun u ↦ ?_
+  refine' ⟨u.image Int.natAbs, fun v' hv' ↦ _⟩
+  let u1 := v'.image fun x : ℕ ↦ (x : ℤ)
+  let u2 := v'.image fun x : ℕ ↦ -(x : ℤ)
+  have A : u ⊆ u1 ∪ u2 := by
+    intro x hx
+    simp only [u1, u2, mem_union, mem_image, exists_prop]
+    rcases le_total 0 x with (h'x | h'x)
+    · left
+      refine' ⟨Int.natAbs x, hv' _, _⟩
+      · simp only [mem_image, exists_prop]
+        exact ⟨x, hx, rfl⟩
+      · simp only [h'x, Int.coe_natAbs, abs_eq_self]
+    · right
+      refine' ⟨Int.natAbs x, hv' _, _⟩
+      · simp only [mem_image, exists_prop]
+        exact ⟨x, hx, rfl⟩
+      · simp only [abs_of_nonpos h'x, Int.coe_natAbs, neg_neg]
+  refine' ⟨u1 ∪ u2, A, _⟩
+  calc
+    (∑ x in u1 ∪ u2, (f x + ite (x = 0) (f 0) 0)) =
+        (∑ x in u1 ∪ u2, f x) + ∑ x in u1 ∩ u2, f x := by
+      rw [sum_add_distrib]
+      congr 1
+      refine' (sum_subset_zero_on_sdiff inter_subset_union _ _).symm
+      · intro x hx
+        suffices x ≠ 0 by simp only [this, if_false]
+        rintro rfl
+        simp only [mem_sdiff, mem_union, mem_image, Nat.cast_eq_zero, exists_eq_right, neg_eq_zero,
+          or_self, mem_inter, and_self, and_not_self, u1, u2] at hx
+      · intro x hx
+        simp only [u1, u2, mem_inter, mem_image, exists_prop] at hx
+        have : x = 0 := by
+          apply le_antisymm
+          · rcases hx.2 with ⟨a, _, rfl⟩
+            simp only [Right.neg_nonpos_iff, Nat.cast_nonneg]
+          · rcases hx.1 with ⟨a, _, rfl⟩
+            simp only [Nat.cast_nonneg]
+        simp only [this, eq_self_iff_true, if_true]
+    _ = (∑ x in u1, f x) + ∑ x in u2, f x := sum_union_inter
+    _ = (∑ b in v', f b) + ∑ b in v', f (-b) := by
+      simp only [u1, u2, Nat.cast_inj, imp_self, implies_true, forall_const, sum_image, neg_inj]
+    _ = ∑ b in v', (f b + f (-b)) := sum_add_distrib.symm
+#align has_sum.sum_nat_of_sum_int HasSum.natCast_add_neg_natCast
+
+-- deprecated 2024-03-04
+@[deprecated HasSum.natCast_add_neg_natCast] alias HasSum.sum_nat_of_sum_int :=
+  HasSum.natCast_add_neg_natCast
+
+theorem Summable.natCast_add_neg_natCast {f : ℤ → M} (hf : Summable f) :
+    Summable fun n : ℕ ↦ f n + f (-n) :=
+  hf.hasSum.natCast_add_neg_natCast.summable
+
+lemma tsum_natCast_add_neg_natCast [T2Space M] {f : ℤ → M} (hf : Summable f) :
+    ∑' n : ℕ, (f n + f (-n)) = (∑' n : ℤ, f n) + f 0 :=
+  hf.hasSum.natCast_add_neg_natCast.tsum_eq
+
+theorem HasSum.of_succ_negSucc {f : ℤ → M}
+    (hf₁ : HasSum (fun n : ℕ ↦ f (n + 1)) m) (hf₂ : HasSum (fun n : ℕ ↦ f (Int.negSucc n)) m') :
+    HasSum f (m + f 0 + m') :=
+  haveI : ∀ g : ℕ → M, HasSum (fun k ↦ g (k + 1)) m → HasSum g (m + g 0) :=
+    fun g hg ↦ by simpa using (add_comm _ m ▸ hg.sum_range_add :)
+  (this (fun n ↦ f n) hf₁).nonneg_add_neg hf₂
```

Comment:
It feels weird to me to prove a general statement for a general function `g`, but to apply it just to the specific function `f`. Why not just write it as: `HasSum f (m + f 0)`?